pkg/asset/installconfig/gcp/dns.go

```diff
@@ -29,6 +32,50 @@ func GetPublicZone(ctx context.Context, project, baseDomain string) (*dns.Manage
 	return dnsZone, nil
 }
 
+// ParseZone will attempt to extract the project and zone from a long zone name that
+// includes the full path (i.e. projects/{project-id}/managedZones/{managed-zone-name}). If the data
+// cannot be parsed or does not include this information, the default (provided) values will be returned
+func ParseZone(defaultProject, providedZoneName string) (string, string, error) {
+	zoneName := providedZoneName
+	project := defaultProject
+
+	matched, err := regexp.MatchString(`^projects/\./managedZones/\.`, zoneName)
+	if err != nil {
+		return "", "", err
+	} else {
+		if matched {
+
+			splitZoneName := strings.Split(zoneName, "/")
+			if len(splitZoneName) < 4 {
+				return project, zoneName, fmt.Errorf("invalid zone format: %s", zoneName)
+			} else {
+				return splitZoneName[1], splitZoneName[3], nil
+			}
+		}
+	}
+
+	return project, zoneName, nil
+}
+
+// GetZoneByName returns a DNS managed zone. The zoneName may include the full path
+// to the zone (i.e. https://dns.googleapis.com/dns/), and the provided project will be
+// ignored and the project will be pulled from the zoneName
+func GetZoneByName(ctx context.Context, project, zoneName string) (*dns.ManagedZone, error) {
+	client, err := NewClient(context.TODO())
+	if err != nil {
+		return nil, err
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+	defer cancel()
+
+	dnsZone, err := client.GetDNSZoneByName(ctx, project, zoneName)
+	if err != nil {
+		return nil, err
+	}
+
+	return dnsZone, nil
+}
```

Comment:
GetZoneByName will no longer be needed if we aren't parsing the zones in the manifests generation.